plugins/crypto/openssl/ua_certificategroup_openssl.c

```diff
@@ -839,6 +841,51 @@ UA_CertificateUtils_getSubjectName(UA_ByteString *certificate,
     return UA_STATUSCODE_GOOD;
 }
 
+UA_StatusCode
+UA_CertificateUtils_getThumbprint(UA_ByteString *certificate,
+                                  UA_String *thumbprint) {
+    if(certificate == NULL || thumbprint->length != (SHA1_DIGEST_LENGTH * 2))
+        return UA_STATUSCODE_BADINTERNALERROR;
+
+    UA_StatusCode retval = UA_STATUSCODE_GOOD;
+
+    BIO *certBio = BIO_new_mem_buf(certificate->data, certificate->length);
+    if(!certBio)
+        return UA_STATUSCODE_BADINTERNALERROR;
+
+    X509 *cert = PEM_read_bio_X509(certBio, NULL, 0, NULL);
```

Comment:
This looks rather restrictive. Why not allowing DER here too (like MbedTLS nicely does for us).